### PR TITLE
fixed the issue where .env file is not used to substitute variables d…

### LINKED ISSUE #726: Environment variables not read from .env file for for compose 3
### Summary
Per [this example](https://github.com/aws/amazon-ecs-cli/issues/537#issuecomment-463146433), environment variables defined in an .env file are not read into docker-compose.yml files using compose version 3.

### Description

v2 docker-compose file:

```
version: '2'
services:
  tracker:
    image: X/image:$TAG
```
RESULT: `TAG` read in

v3 docker-compose file:

```
version: '3'
services:
  tracker:
    image: X/image:$TAG
```
RESULT: `TAG` NOT read in

### Expected Behavior
Vars defined in .env would be interpolated into `docker-compose.yml` for both compose versions.

### Observed Behavior
Only v2 compose projects correctly interpolate vars defined in .env

### PR DESCRIPTION
**Pull request by hencrice**
…efined in V3 compose file

This commit fixes #726. The cause for the issue is the code path that handles variables defined in the `.env` file in Compose V3 is totally separated from the ones used by Compose V1, V2 (#839). So prior to this change, there was no code that actually handles reading from the `.env` file for Compose V3.

**Issue #, if available**:
#726 

**Description of changes**:
Implemented logic in parseV3 to handle the `.env` file.

----
Enter `[N/A]` in the box, if an item is not applicable to your change.

**Testing**  
- [X] Unit tests passed
- [X] Integration tests passed
- [X] Unit tests added for new functionality
- [X] Listed manual checks and their outputs in the comments ([example](https://github.com/aws/amazon-ecs-cli/pull/750#issuecomment-472623042))
- [N/A] Link to issue or PR for the integration tests: 

**Documentation**  
- [N/A] Contacted our doc writer
- [N/A] Updated our README
----

**Manual Tests**
Before the code change, the `TAG` variable is not resolved for V3 Compose file:

```
$ cat docker-compose.yml
version: '3'
services:
  web:
    image: amazon/amazon-ecs-sample:$TAG
    ports:
      - "80:80"
    stop_grace_period: 1m15s

$ cat .env
TAG=1
```

```
$ aws ecs describe-task-definition --task-definition ecs-cli-tutorial:7 --region us-west-2
{
    "taskDefinition": {
        "taskDefinitionArn": "arn:aws:ecs:us-west-2:620297136107:task-definition/ecs-cli-tutorial:7",
        "containerDefinitions": [
            {
                "name": "web",
                "image": "amazon/amazon-ecs-sample:",
                "cpu": 0,
                "links": [],
                "portMappings": [
                    {
                        "containerPort": 80,
                        "hostPort": 80,
                        "protocol": "tcp"
                    }
                ],
                "essential": true,
                "entryPoint": [],
                "command": [],
                "environment": [],
                "mountPoints": [],
                "volumesFrom": [],
                "linuxParameters": {
                    "capabilities": {},
                    "devices": []
                },
                "stopTimeout": 75,
                "privileged": false,
                "readonlyRootFilesystem": false,
                "dnsServers": [],
                "dnsSearchDomains": [],
                "extraHosts": [],
                "dockerSecurityOptions": [],
                "pseudoTerminal": false
            }
        ],
        "family": "ecs-cli-tutorial",
        "executionRoleArn": "arn:aws:iam::620297136107:role/ecsTaskExecutionRole",
        "networkMode": "awsvpc",
        "revision": 7,
        "volumes": [],
        "status": "ACTIVE",
        "requiresAttributes": [
            {
                "name": "com.amazonaws.ecs.capability.docker-remote-api.1.17"
            },
            {
                "name": "ecs.capability.container-ordering"
            },
            {
                "name": "com.amazonaws.ecs.capability.docker-remote-api.1.18"
            },
            {
                "name": "ecs.capability.task-eni"
            }
        ],
        "placementConstraints": [],
        "compatibilities": [
            "EC2",
            "FARGATE"
        ],
        "requiresCompatibilities": [
            "EC2"
        ],
        "cpu": "256",
        "memory": "512"
    }
}
```

After the code change:
```
$ cat .env
TAG=01
```

The `TAG` variable is correctly resolved for V3:
```
$ aws ecs describe-task-definition --task-definition ecs-cli-tutorial:11 --region us-west-2
{
    "taskDefinition": {
        "taskDefinitionArn": "arn:aws:ecs:us-west-2:620297136107:task-definition/ecs-cli-tutorial:11",
        "containerDefinitions": [
            {
                "name": "web",
                "image": "amazon/amazon-ecs-sample:01",
                "cpu": 0,
                "links": [],
                "portMappings": [
                    {
                        "containerPort": 80,
                        "hostPort": 80,
                        "protocol": "tcp"
                    }
                ],
                "essential": true,
                "entryPoint": [],
                "command": [],
                "environment": [],
                "mountPoints": [],
                "volumesFrom": [],
                "linuxParameters": {
                    "capabilities": {},
                    "devices": []
                },
                "stopTimeout": 75,
                "privileged": false,
                "readonlyRootFilesystem": false,
                "dnsServers": [],
                "dnsSearchDomains": [],
                "extraHosts": [],
                "dockerSecurityOptions": [],
                "pseudoTerminal": false
            }
        ],
        "family": "ecs-cli-tutorial",
        "executionRoleArn": "arn:aws:iam::620297136107:role/ecsTaskExecutionRole",
        "networkMode": "awsvpc",
        "revision": 11,
        "volumes": [],
        "status": "ACTIVE",
        "requiresAttributes": [
            {
                "name": "com.amazonaws.ecs.capability.docker-remote-api.1.17"
            },
            {
                "name": "ecs.capability.container-ordering"
            },
            {
                "name": "com.amazonaws.ecs.capability.docker-remote-api.1.18"
            },
            {
                "name": "ecs.capability.task-eni"
            }
        ],
        "placementConstraints": [],
        "compatibilities": [
            "EC2",
            "FARGATE"
        ],
        "requiresCompatibilities": [
            "EC2"
        ],
        "cpu": "256",
        "memory": "512"
    }
}
```

The `TAG` variable is still correctly resolved for V2 (i.e. no regression):
```
$ aws ecs describe-task-definition --task-definition ecs-cli-tutorial:12 --region us-west-2
{
    "taskDefinition": {
        "taskDefinitionArn": "arn:aws:ecs:us-west-2:620297136107:task-definition/ecs-cli-tutorial:12",
        "containerDefinitions": [
            {
                "name": "web",
                "image": "amazon/amazon-ecs-sample:01",
                "cpu": 0,
                "links": [],
                "portMappings": [
                    {
                        "containerPort": 80,
                        "hostPort": 80,
                        "protocol": "tcp"
                    }
                ],
                "essential": true,
                "entryPoint": [],
                "command": [],
                "environment": [],
                "mountPoints": [],
                "volumesFrom": [],
                "linuxParameters": {
                    "capabilities": {},
                    "devices": []
                },
                "stopTimeout": 75,
                "privileged": false,
                "readonlyRootFilesystem": false,
                "dnsServers": [],
                "dnsSearchDomains": [],
                "extraHosts": [],
                "dockerSecurityOptions": [],
                "pseudoTerminal": false,
                "dockerLabels": {},
                "ulimits": []
            }
        ],
        "family": "ecs-cli-tutorial",
        "executionRoleArn": "arn:aws:iam::620297136107:role/ecsTaskExecutionRole",
        "networkMode": "awsvpc",
        "revision": 12,
        "volumes": [],
        "status": "ACTIVE",
        "requiresAttributes": [
            {
                "name": "com.amazonaws.ecs.capability.docker-remote-api.1.17"
            },
            {
                "name": "ecs.capability.container-ordering"
            },
            {
                "name": "com.amazonaws.ecs.capability.docker-remote-api.1.18"
            },
            {
                "name": "ecs.capability.task-eni"
            }
        ],
        "placementConstraints": [],
        "compatibilities": [
            "EC2",
            "FARGATE"
        ],
        "requiresCompatibilities": [
            "EC2"
        ],
        "cpu": "256",
        "memory": "512"
    }
}
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: ecs-cli/modules/cli/compose/project/project_parseV3_test.go [hunk 1]
```diff
@@ -509,6 +510,50 @@ func TestThrowErrorIfFileDoesNotExist(t *testing.T) {
 	assert.Error(t, err)
 }
 
+func TestParseV3WithDefaultDotEnvFile(t *testing.T) {
+	// Set up the .env file
+	dir, err := ioutil.TempDir("", "dotEnvTest")
+	assert.NoError(t, err, "Unexpected error in creating temp directory that will hold the .env file")
```
Comment (iamhopaul123) [inline ecs-cli/modules/cli/compose/project/project_parseV3_test.go]: Nitpick: error message should not be capitalized.

Comment (hencrice) [reply ecs-cli/modules/cli/compose/project/project_parseV3_test.go]: oh this is the message that will be printed out when an assertion fails, not a customer-facing error message emitted when CLI is being used. Also, I followed the formatting based on the other test cases.

File: ecs-cli/modules/cli/compose/project/project_parseV3.go [hunk 2]
```diff
@@ -270,8 +273,21 @@ func logWarningForDeployFields(d types.DeployConfig, serviceName string) {
 	}
 }
 
-func getEnvironment() map[string]string {
-	env := os.Environ()
+func getEnvironment(wrkDir string) map[string]string {
+	env, err := opts.ParseEnvFile(filepath.Join(wrkDir, dotEnvFileName))
```
Comment (SoManyHs) [inline ecs-cli/modules/cli/compose/project/project_parseV3.go]: What happens if the .env file doesn't exist?

Comment (hencrice) [reply ecs-cli/modules/cli/compose/project/project_parseV3.go]: oh, then a file-not-exist err will be returned by `ParseEnvFile` and ignored by ln 284

Comment (SoManyHs) [reply ecs-cli/modules/cli/compose/project/project_parseV3.go]: Ah right, didn't see that. one way we've checked for this pre-emptively is using `os.Stat`, but this works fine too.

**Pull request merged**